docs/dev/releases.md

```diff
@@ -0,0 +1,102 @@
+# Releases
+
+This doc explains how to build, test and release ingress controllers.
+
+## Building
+
+All ingress controllers are build through a Makefile. Depending on your
+requirements you can build a raw server binary, a local container image,
+or push an image to a remote repository.
+
+Build a raw server binary
+```console
+$ make controller
+```
+
+Build a local container image
+```console
+$ make container
+```
+
+Push the container image to a remote repository
+```console
+$ make push TAG=0.0 PREFIX=$USER/ingress-controller
```

Comment:
I think we'll have to udpate existing Makefiles for consistency, then update this doc with whiever convention we pick. Planning to do that in an iteration.